functions/src/src/AddEvent.ts

```diff
@@ -0,0 +1,29 @@
+import * as admin from 'firebase-admin'
+import { getUser } from './GetUser'
+import { UserPermissionLevel } from '../models/UserPermissionLevel'
+import { APIResponse } from '../models/APIResponse'
+import { HouseCompetition } from '../models/HouseCompetition'
+import { Event } from '../models/Event'
+
+export async function addEvent(event: Event): Promise<Boolean> {
+    
+    let user = await getUser(event.creator_id)
```

Comment:
Can you take user:User as a function parameter instead of retrieving it with getUser. This will help make this function more reusable for other endpoints by avoid having to retrieve the user model again.

Reply:
Ignore this comment as I explain in the overall review comment a better format for endpoint and function.


But sneak peak, don't take User or Event as input params. Take the basic data types for event so you can verify the validity of the data, then create event and post to dB in this function. 

Also, please add point type id

Reply:
Although, for further clarification, for any function that creates an instance of a model, you would want to take basic parameters. But if you were writing a method that modified an instance of a model or checked the validity of the model, you would want to take the model as input parameters. 

The general concept being, if you are creating a model, take basic types. 

If you are using or modifying an existing model, take in that model as a parameters so that the function can be more efficiently used by other endpoints if need be.